src/Integration/Connection/ConnectionController.cs

```diff
@@ -109,37 +97,43 @@ internal bool IsConnectionInProgress
                 }
             }
         }
+
         #endregion
 
         #region Connect Command
 
         private bool CanConnect(ConnectConfiguration configuration)
         {
             return solutionInfoProvider.IsSolutionFullyOpened()
-                && !this.host.VisualStateManager.IsConnected
-                && !this.host.VisualStateManager.IsBusy;
+                   && !this.host.VisualStateManager.IsConnected
+                   && !this.host.VisualStateManager.IsBusy;
         }
 
         private void OnConnect(ConnectConfiguration configuration)
         {
             Debug.Assert(this.CanConnect(configuration));
             Debug.Assert(!this.host.VisualStateManager.IsBusy, "Service is in a connecting state");
-           
+
             ConnectionInformation connectionInfo;
 
             var sharedConfig = host.SharedBindingConfig;
             var credentials = host.GetCredentialsForSharedConfig();
             var useSharedConfig = configuration?.UseSharedBinding ?? false;
             var autoBindProjectKey = host.VisualStateManager.BoundProjectKey;
 
-            if (useSharedConfig && !string.IsNullOrEmpty(autoBindProjectKey) && sharedConfig != null && credentials != null)
+            if (useSharedConfig && !string.IsNullOrEmpty(autoBindProjectKey) && sharedConfig != null)
             {
                 connectionInfo = new ConnectionInformation(sharedConfig.ServerUri,
-                    credentials.Username,
-                    credentials.Password.ToSecureString())
+                    credentials?.Username,
+                    credentials?.Password?.ToSecureString())
                 {
                     Organization = new SonarQubeOrganization(sharedConfig.Organization, string.Empty)
                 };
+
+                if (credentials == null)
```

Comment:
Maybe move this up to prevent creating connectioninfo unnecessarily

Reply:
It's always used, no? sometimes in line 135, and always in line 145

Reply:
sometimes = conditionally